lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -289,6 +289,48 @@ class DecomposeAtenExpandOp : public OpRewritePattern<AtenExpandOp> {
 };
 } // namespace
 
+namespace {
+class DecomposeAtenAddCMulOp : public OpRewritePattern<AtenAddCMulOp> {
```

Comment:
You should be able to use a template here like you did in `RefineTypes`:

```cpp
template<typename OpTy, typename T1T2Op>
class DecomposeAtenAddCLikeOp : public OpRewritePattern<OpTy> {
  using OpRewritePattern::OpRewritePattern;
  LogicalResult matchAndRewrite(OpTy op,
                                PatternRewriter &rewriter) const override {
    // some code

    Value product = rewriter.create<T1T2Op>(loc, tensorType, t1, t2);
    
  // more code
  }
};
```

Then when you add the pattern, you just do

```cpp
patterns.add<DecomposeAtenAddCLikeOp<AtenAddCMulOp, AtenMulTensorOp>>(context);
// same for div
```